diff ./CMakeLists.txt ./CMakeLists.txt
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -85,7 +85,9 @@
 if(PNG_HARDWARE_OPTIMIZATIONS)
 # set definitions and sources for arm
 if(CMAKE_SYSTEM_PROCESSOR MATCHES "^arm" OR
-  CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64")
+  CMAKE_SYSTEM_PROCESSOR MATCHES "^aarch64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "^arm64"
+ OR(${PNG_ARM_NEON} STREQUAL "on")
+       )
   set(PNG_ARM_NEON_POSSIBLE_VALUES check on off)
   set(PNG_ARM_NEON "check" CACHE STRING "Enable ARM NEON optimizations:
      check: (default) use internal checking code;



